### PR TITLE
[WIP] Remove table layout handle

### PR DESCRIPTION
**Pull request by hellium01**
This PR removes tableLayoutHandle from planNode. In general, we should remove the requirement of the layout, I will submit another PR later but this is what we needed to clean up planNode so far.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/metadata/TableHandle.java [hunk 1]
```diff
@@ -48,29 +58,48 @@ public ConnectorTableHandle getConnectorHandle()
...
+    public int hashCode()
+    {
+        return Objects.hash(connectorId, connectorHandle, transaction, layout);
     }
 
     @Override
     public String toString()
     {
-        return connectorId + ":" + connectorHandle;
+        return "TableHandle{" +
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/metadata/TableHandle.java]: Maybe a space before "{"?

File: presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java [hunk 2]
```diff
@@ -1216,6 +1217,16 @@ private boolean canResolveOperator(OperatorType operatorType, List<TypeSignature
         }
     }
 
+    private ConnectorTableLayoutHandle getTableLayout(Session session, TableHandle tableHandle)
+    {
+        if (tableHandle.getLayout().isPresent()) {
+            return tableHandle.getLayout().get();
+        }
+        Optional<TableLayoutResult> result = getLayout(session, tableHandle, alwaysTrue(), Optional.empty());
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java]: The function names here start to get really confusing.
* There are two `getLayout` now but they return different types. Maybe rename this one `getLayoutResult`.
* THere are two `getTableLayout` and the `ConnectorMetadata.getTableLayout` returns a `ConnectorTableLayout` while this function returns `ConnectorTableLayoutHandle`. Maybe just name this function `getConnectorTableLayoutHandle`.

File: presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java [hunk 3]
```diff
@@ -15,10 +15,21 @@
 
 import com.facebook.presto.spi.ConnectorSession;
 import com.facebook.presto.spi.ConnectorSplitSource;
+import com.facebook.presto.spi.ConnectorTableHandle;
 import com.facebook.presto.spi.ConnectorTableLayoutHandle;
 
 public interface ConnectorSplitManager
 {
+    default ConnectorSplitSource getSplits(
+            ConnectorTableHandle tableHandle,
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java]: The parameter order of the old `getSplits` makes more sense to me.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java [hunk 4]
```diff
@@ -55,35 +51,32 @@ public TableScanNode(
...
         this.outputSymbols = ImmutableList.copyOf(requireNonNull(outputs, "outputs is null"));
         this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
         checkArgument(assignments.keySet().containsAll(outputs), "assignments does not cover all of outputs");
-        this.tableLayout = requireNonNull(tableLayout, "tableLayout is null");
         this.currentConstraint = null;
         this.enforcedConstraint = null;
     }
 
-    public TableScanNode(
+    public static TableScanNode createTableScanNode(
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java]: It's a little weird to have one create as a static function and the other directly using constructor. Two constructors are fine I think.

File: presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java [hunk 5]
```diff
@@ -67,6 +67,14 @@ default boolean schemaExists(ConnectorSession session, String schemaName)
         return listSchemaNames(session).contains(schemaName);
     }
 
+    /**
+     * Check if connector is providing legacy connnectorTableLayout
+     */
+    default boolean useLegacyLayout()
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java]: Can we introduce this function together with the logic using it?

File: presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java (hunk 2 above)
Comment (hellium01) [reply presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java]: Yes, TableLayout will be renamed to TableProperties later. ConnectorTableLayout will become ConnectorTableProperties.

File: presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java (hunk 3 above)
Comment (hellium01) [reply presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java]: The idea is to use tableHandle represent connector's view of table (physical positions, enforced predicates(what today's Layout handles), ... etc.) and use trait/property to represent all these constraints (tupledomain, ...), but we are not there yet.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java (hunk 4 above)
Comment (hellium01) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TableScanNode.java]: Yeah, but there is two constructor with same parameter.

File: presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java (hunk 5 above)
Comment (hellium01) [reply presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java]: Yes, this is added just as placeholder, will remove later.

**Pull request closed without merging**